File: src/main/bin/perl/OSConfig.pl
#!/usr/bin/perl
use FindBin '$RealBin';
use File::Basename qw(dirname);
use Cwd  qw(abs_path);
use lib dirname(dirname abs_path $0).'/bin';
use osversion;
use Common;

#sometimes raw module might not be loaded into kernel. run "modprobe raw" anyway
my $my_script_dir = $RealBin;
my $log = Log::Log4perl->get_logger("OSConfig");
$log->debug("starting");
if (osversion::is_ubuntu()) {
    my $cmd = "modprobe raw";
    (system($cmd) == 0) or exitProcess(__FILE__, __LINE__, "fail to modprobe raw module to the kernel");
    $log->debug("modprobe raw module is loaded to the kernel");
    sleep 2;
}


# change the max map count for pages
$cmd = "echo 67107840 > /proc/sys/vm/max_map_count";
(system($cmd) == 0) or exitProcess(__FILE__, __LINE__, "fail to change max_map_count, cmd:$cmd");
$log->debug("finished");

File: src/main/bin/os_config.py
# -*- encoding=utf-8 -*-
# Copyright (C) 2013-2024 Nanjing Pengyun Network Technology Co., Ltd.
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# 
"""
copy from OSConfig.pl
import Common
import os_version
import time
from utils import logging_config
from utils.common import exit_process

logger = logging_config.get_logger()

if os_version.is_ubuntu():
  ret_code, ret_line = Common.run_cmd("modprobe raw")
  if ret_code != 0:
    exit_process(-1, "fail to modprobe raw module to the kernel")

  logger.info("modprobe raw module is loaded to the kernel")
  time.sleep(2)

ret_code, ret_line = Common.run_cmd(
    "echo 67107840 > /proc/sys/vm/max_map_count")
if ret_code != 0:
  exit_process(-1, "fail to change max_map_count")
